StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts

```diff
@@ -18,7 +18,7 @@ import { TokensService } from '../../services/tokens.service';
 })
 @Mixin([Disposable])
 export class AddTokenComponent implements OnInit, OnDestroy, Disposable {
-
+  private subscriptions: Subscription[] = [];
```

Comment:
Minor suggestion, change this to `private subscriptions: Subscription`.

Then add all subscriptions to it further down buy using `this.subscriptions.add(someNewSub$);`

This allows all subscriptions to be held together without having to deal with an array.

Then onDestory all you do is `this.subscriptions.unsubscribe();` maybe `if (this.subscriptions) this.subscriptions.unsubscribe()` if there are no values for whatever reason.

If you curious about this practice or others, you can read about it here https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f.

Just a suggestion, could probably make some other components logic simpler.

Reply:
Sure, noted, more of a personal preference and consistency I think. Problem is sometimes subscriptions are made conditionally at runtime, so do you keep checking if it's null then create or add, do you just create a dummy root subscription? Will leave as is for now but will play with this in the future.